src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -191,30 +201,67 @@ const std::shared_ptr<AbstractOperator>& SQLPipelineStatement::get_physical_plan
   return _physical_plan;
 }
 
-const std::vector<std::shared_ptr<OperatorTask>>& SQLPipelineStatement::get_tasks() {
+const std::vector<std::shared_ptr<AbstractTask>>& SQLPipelineStatement::get_tasks() {
   if (!_tasks.empty()) {
     return _tasks;
   }
 
-  _tasks = OperatorTask::make_tasks_from_operator(get_physical_plan(), _cleanup_temporaries);
+  if (_is_transaction_statement) {
+    _tasks = _get_transaction_tasks();
+  } else {
+    auto operator_tasks = OperatorTask::make_tasks_from_operator(get_physical_plan(), _cleanup_temporaries);
+    _tasks = std::vector<std::shared_ptr<AbstractTask>>(operator_tasks.cbegin(), operator_tasks.cend());
+  }
+
   return _tasks;
 }
 
+std::vector<std::shared_ptr<AbstractTask>> SQLPipelineStatement::_get_transaction_tasks() {
```

Comment:
I just tried to change it. The return value is stored into `std::vector<std::shared_ptr<AbstractTask>>` by the calling function `SQLPipelineStatement::get_tasks()`. This means, the return value would have to get implicitly converted from `std::vector<std::shared_ptr<JobTask>>` to `std::vector<std::shared_ptr<AbstractTask>>`. However, it turns out, that vectors cannot do this kind of conversion implicitly. I guess it would be fine, if the returned Task weren't encapsulated inside a vector.